### ISSUE TITLE
font shadow missing in exported png file

### ISSUE THREAD

**Issue by geyanku** [meltingice/psd.js#57]
I have a psd file, with one font layer. When exporting png file, the font shadow is missing. Here's my code:
PSD = require('psd');

PSD.open('psd/1.psd').then(function(psd) {
  psd.tree().descendants().forEach(function(node) {
    if(node.isGroup()) return true;
    node.saveAsPng("./output1/" + node.name + ".png").catch(function(err) {
      console.log(err.stack);
    }); 
  }); 
}).then(function() {
  console.log('finished');
});

Help me please

**Comment by lordminor** [comment]
I think the problem is that you are trying to export only the layer. The same happen when you try to render every layer in separate image, the fx doesn't render. (strokes, shadows, and others..)

Try if you export the whole psd file, to see if is correctly rendered.

**Comment by meltingice** [comment]
Yeah, text shadow is not saved with the image data unfortunately. It's only stored as a layer effect, which must be rendered manually. PSD.js does not support this, yet.

--
Ryan LeFevre (@meltingice)
Sr. Software Engineer
HODINKEE




On Sun, May 1, 2016 at 3:25 PM -0700, "lordminor" <notifications@github.com> wrote:












I think the problem is that you are trying to export only the layer. The same happen when you try to render every layer in separate image, the fx doesn't render. (strokes, shadows, and others..)



Try if you export the whole psd file, to see if is correctly rendered.
